### PR TITLE
chore(sdk): Prepare SDK types for asset identifier refactoring

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/sdk/src/swap/assets.ts [hunk 1]
```diff
@@ -31,13 +32,13 @@ const assetFactory =
...
     return {
-      id: asset,
+      asset,
       chain: assetChains[asset],
       contractAddress: getTokenContractAddress(asset, network, false),
       decimals: assetDecimals[asset],
       name: assetName[asset],
       symbol: asset,
-      chainflipId: asset,
+      chainflipId: toCamelCase(asset),
```
Comment (niklasnatter) [inline packages/sdk/src/swap/assets.ts:41]: todo: revert this in https://github.com/chainflip-io/chainflip-sdk-monorepo/pull/550

File: packages/sdk/src/swap/services/ApiService.ts [hunk 2]
```diff
@@ -59,14 +59,16 @@ const getQuote: BackendQuery<
...
     amount: returnedRequestData.amount,
+    srcChain: returnedRequestData.srcChain,
     srcAsset: returnedRequestData.srcAsset,
+    destChain: returnedRequestData.destChain,
     destAsset: returnedRequestData.destAsset,
     ...(brokerCommissionBps && {
       brokerCommissionBps: String(brokerCommissionBps),
     }),
-  };
+  } as QuoteQueryParams;
```
Comment (niklasnatter) [inline packages/sdk/src/swap/services/ApiService.ts:71]: todo: revert this in https://github.com/chainflip-io/chainflip-sdk-monorepo/pull/550

**Review by jerryafr — APPROVED** [review]

File: packages/shared/src/strings.ts [hunk 3]
```diff
@@ -5,6 +5,10 @@ export type CamelCaseToSnakeCase<S extends string> =
         : ''}${Lowercase<T>}${CamelCaseToSnakeCase<U>}`
     : S;
 
+type CamelCase<T extends string> = T extends `${infer F}${infer R}`
```
Comment (GabrielBuragev) [inline packages/shared/src/strings.ts:8]: isnt this PascalCase?

**Review by zoheb391 — APPROVED** [review]

**Review by GabrielBuragev — APPROVED** [review]

File: packages/sdk/src/swap/assets.ts (hunk 1 above)
Comment (acdibble) [inline packages/sdk/src/swap/assets.ts:41]:
```suggestion
      chainflipId: screamingSnakeToPascalCase(asset),
```

File: packages/shared/src/strings.ts (hunk 3 above)
Comment (niklasnatter) [reply packages/shared/src/strings.ts:8]: i have no idea lol, copied the type from another repo. will change

File: packages/sdk/src/swap/assets.ts (hunk 1 above)
Comment (niklasnatter) [reply packages/sdk/src/swap/assets.ts]: oh nice, thats even better!

**Pull request merged**